src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/GroupTransformers/XamlIncludeGroupTransformer.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Avalonia.Markup.Xaml.XamlIl.CompilerExtensions.Transformers;
+using Avalonia.Platform;
+using XamlX;
+using XamlX.Ast;
+using XamlX.Emit;
+using XamlX.IL;
+using XamlX.Transform;
+using XamlX.Transform.Transformers;
+using XamlX.TypeSystem;
+
+namespace Avalonia.Markup.Xaml.XamlIl.CompilerExtensions.GroupTransformers;
+
+internal class AvaloniaXamlIncludeTransformer : IXamlAstGroupTransformer
+{
+    public IXamlAstNode Transform(AstGroupTransformationContext context, IXamlAstNode node)
+    {
+        if (node is not XamlValueWithManipulationNode valueNode
+            || valueNode.Value is not XamlAstNewClrObjectNode objectNode
+            || (objectNode.Type.GetClrType() != context.GetAvaloniaTypes().StyleInclude
+                && objectNode.Type.GetClrType() != context.GetAvaloniaTypes().ResourceInclude))
+        {
+            return node;
+        }
+
+        var nodeTypeName = objectNode.Type.GetClrType().Name;
+        if (valueNode.Manipulation is not XamlObjectInitializationNode
+            {
+                Manipulation: XamlPropertyAssignmentNode { Property: { Name: "Source" } } sourceProperty
+            })
+        {
+            return context.ParseError($"Source property must be set on the \"{nodeTypeName}\" node.", node);
+        }
+
+        if (sourceProperty.Values.OfType<XamlAstNewClrObjectNode>().FirstOrDefault() is not { } sourceUriNode
+            || sourceUriNode.Type.GetClrType() != context.GetAvaloniaTypes().Uri
+            || sourceUriNode.Arguments.FirstOrDefault() is not XamlConstantNode { Constant: string originalAssetPath })
+        {
+            // TODO: make it a compiler warning
+            // Source value can be set with markup extension instead of the Uri object node, we don't support it here yet.
+            return node;
+        }
+
+        if (originalAssetPath.StartsWith("avares://"))
```

Comment:
I think it would be more readable as
```cs
if(originalAssetPath.StartsWith("/"))
   ...
   
if(!originalAssetPath.StartsWith("avares://"))
   throw ...
```